stl/inc/experimental/random

```diff
@@ -0,0 +1,55 @@
+// random experimental header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _EXPERIMENTAL_RANDOM_
+#define _EXPERIMENTAL_RANDOM_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#include <random>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+namespace experimental {
+    inline namespace fundamentals_v3 {
+
+        inline default_random_engine& _Engine() {
+            thread_local default_random_engine _Eng{random_device{}()};

```

Comment:
I am unsure about that but do you need _STD qualifier here for the random_device

Reply:
Probably, my understanding of ADL is... poop. I guess I'll wait for a maintainer to comment.

Reply:
Correct, ADL applies to functions, not classes (or alias templates, variable templates, etc.). As long as you're within `std` or a sub-namespace like `std::experimental`, you don't need to `_STD` qualify non-function identifiers. Unqualified name lookup is guaranteed to find the identifier you want in `std` before considering anything in the global namespace (or anything that's been dragged into the global namespace by a `using`-directive or `using`-declaration from the user).

The only places in the codebase where you'll see `_STD` qualified non-function identifiers are when we aren't within `std`, or there's potential ambiguity between namespace-scope and class-scope identifiers (`reverse_iterator` is a notable example).